### DISCUSSION TITLE
How to set/change settings programatically?

### DISCUSSION THREAD

**Discussion by TangrisJones** [Mikubill/sd-webui-controlnet#685] [Q&A]
Hi, sorry if this is already covered somewhere. 

I'm interested in spinning this thing up in a docker container on some server & then leverage the SD webui API for an app I'm building. Does anyone know how to go about changing the default settings programatically? The most important setting I want to change is the maximum control net models to cache (AKA the `control_net_model_cache_size` as seen [here](https://github.com/Mikubill/sd-webui-controlnet/search?q=control_net_model_cache_size)). Ideally I want to increase it automatically to 5. 

Thanks for the help! :D

**Comment by ljleb** [comment]
The settings can be found in `ui-config.json`: https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Features#ui-config

You have to save them at least once to see them I think, but once you have to file created you can just add or replace the controlnet config values.
